plugins/CoreHome/vue/src/WidgetLoader/WidgetLoader.vue

```diff
@@ -0,0 +1,227 @@
+          target="_blank"
+          href="https://matomo.org/faq/troubleshooting/faq_19489/"
+          v-if="hasErrorFaqLink"
+        >
+          {{ translate('General_ErrorRequestFaqLink') }}
+        </a>
+      </div>
+    </div>
+    <div class="theWidgetContent" ref="widgetContent" />
+  </div>
+</template>
+
+<script lang="ts">
+import { IRootScopeService } from 'angular';
+import { defineComponent } from 'vue';
+import ActivityIndicator from '../ActivityIndicator/ActivityIndicator.vue';
+import translate from '../translate';
+import Matomo from '../Matomo/Matomo';
+import AjaxHelper from '../AjaxHelper/AjaxHelper';
+import { NotificationsStore } from '../Notification';
+import MatomoUrl from '../MatomoUrl/MatomoUrl';
+
+/**
+ * Loads any custom widget or URL based on the given parameters.
+ *
+ * The currently active idSite, period, date and segment (if needed) is automatically
+ * appended to the parameters. If this widget is removed from the DOM and requests are in
+ * progress, these requests will be aborted. A loading message or an error message on failure
+ * is shown as well. It's kinda similar to ng-include but there it is not possible to
+ * listen to HTTP errors etc.
+ *
+ * Example:
+ * <WidgetLoader :widget-params="{module: '', action: '', ...}"/>
+ */
+export default defineComponent({
+  props: {
+    widgetParams: Object,
+    widgetName: String,
+  },
+  components: {
+    ActivityIndicator,
+  },
+  data() {
+    return {
+      loading: false,
+      loadingFailed: '',
+      changeCounter: 0, // TODO: check that there is no rerender here?
+      lastWidgetRequest: null,
+      currentScope: null,
+    };
+  },
+  watch: {
+    widgetParams(parameters: Record<string, unknown>) {
+      if (parameters) {
+        this.loadWidgetUrl(parameters, this.changeCounter += 1);
+      }
+    },
+  },
+  computed: {
+    loadingMessage() {
+      if (!this.widgetName) {
+        return translate('General_LoadingData');
+      }
+
+      return translate('General_LoadingPopover', this.widgetName);
+    },
+    hasErrorFaqLink() {
+      const isGeneralSettingsAdminEnabled = Matomo.config.enable_general_settings_admin;
+      const isPluginsAdminEnabled = Matomo.config.enable_plugins_admin;
+
+      return Matomo.hasSuperUserAccess
+        && (isGeneralSettingsAdminEnabled
+          || isPluginsAdminEnabled);
+    },
+  },
+  unmounted() {
+    this.cleanupLastWidgetContent();
+  },
+  methods: {
+    abortHttpRequestIfNeeded() {
+      if (this.lastWidgetRequest) {
+        this.lastWidgetRequest.abort();
+        this.lastWidgetRequest = null;
+      }
+    },
+    cleanupLastWidgetContent() {
+      const { widgetContent } = this.$refs;
+      if (widgetContent) {
+        widgetContent.innerHTML = '';
+      }
+      if (this.currentScope) {
+        this.currentScope.$destroy();
+      }
+    },
+    getWidgetUrl(parameters: Record<string, unknown>): Record<string, unknown> {
+      // TODO: test this
```

Comment:
do we still want to add a test for this?

Reply:
thanks for catching that, this was a reminder to do a manual test. I'll do that and remove the comment.